satpy/readers/netcdf_utils.py

```diff
@@ -431,3 +434,205 @@ def _get_attr(self, obj, key):
+        self.preload_step = preload_step
+        if preload:
+            if not isinstance(ref_fh, BaseFileHandler):
+                raise TypeError(
+                    "Expect reference filehandler when preloading, got "
+                    f"{type(ref_fh)!s}")
+            self.ref_fh = ref_fh
+            if not isinstance(rc_cache, (str, bytes, os.PathLike)):
+                raise TypeError(
+                    "Expect cache file when preloading, got "
+                    f"{type(rc_cache)!s}")
+            self.rc_cache = rc_cache
+        super().__init__(*args, **kwargs)
+        if "required_netcdf_variables" not in self.filetype_info:
+            raise ValueError("For preloadable filehandlers, "
+                             "required_netcdf_variables is mandatory.")
+
+    def _collect_listed_variables(self, file_handle, listed_variables):
+        """Collect listed variables, either preloaded or regular."""
+        if self.preload:
+            self._preload_listed_variables(listed_variables)
+        else:
+            super()._collect_listed_variables(file_handle, listed_variables)
+
+    def _preload_listed_variables(self, listed_variables):
+        """Preload listed variables, either from RC cache or segment cache."""
+        variable_name_replacements = self.filetype_info.get("variable_name_replacements")
+        with open(self.rc_cache, "rb") as fp:
+            self.file_content.update(pickle.load(fp))  # nosec
+        for raw_name in listed_variables:
+            for subst_name in self._get_required_variable_names(
+                    [raw_name], variable_name_replacements):
+                if self._can_get_from_other_segment(raw_name):
+                    self.file_content[subst_name] = self.ref_fh[subst_name]
+                elif not self._can_get_from_other_rc(raw_name):
+                    self._collect_variable_delayed(subst_name)
+
+    def _can_get_from_other_segment(self, itm):
+        """Return true if variable is segment-cachable."""
+        return "segment" in self.filetype_info["required_netcdf_variables"][itm]
+
+    def _can_get_from_other_rc(self, itm):
+        """Return true if variable is rc-cachable."""
+        # it's always safe to store variable attributes, shapes, dtype, dimensions
+        # between repeat cycles
+        if self._is_safely_shareable_metadata(itm):
+            return True
+        # need an inverted mapping so I can tell which variables I can store
+        invmap = self._get_inv_name_map()
+
+#        listed_variables = self.filetype_info.get("required_netcdf_variables")
+#        variable_name_replacements = self.filetype_info.get("variable_name_replacements")
+#        invmap = {}
+#        for raw_name in listed_variables:
+#            for subst_name in self._get_required_variable_names([raw_name],
+#                                                                variable_name_replacements):
+#                invmap[subst_name] = raw_name
```

Comment:
Is this needed?

Reply:
Oops, no, of course this is not needed...  Forgot to push my final commit (now pushed and fixed in fd4b225c2).